### ISSUE TITLE
Option check operator status of transfer departments

### ISSUE THREAD

**Issue by ipso** [LiveHelperChat/livehelperchat#745]
We have certain "tier 1" support staff that should take most of the chats, however in the event that they are all busy or don't answer chats quick enough, we want to automatically transfer chats to "tier 2" support staff. 

In order to achieve this, we setup a "Support" department for "tier 1" and a hidden "Support (Tier 2)" department for tier 2, then setup the "Support" department "Chat transfer workflow" to transfer to "Support (Tier 2)" after 30 seconds. 

This all works great, except after hours where no tier 1 operators are logged in, but tier 2 operators are, the Start Chat form shows the users that the "Support" department is offline. Technically it is I guess, but its setup to transfer to another department which is online, so shouldn't it check that department as well, or at least have an option to do so?

**Comment by remdex** [comment]
Well, perhaps settings online hours in department would do the trick you wish for. But in any case never though of this scenario :) Just how embed part looks like? you can remove your domain from this.

**Comment by ipso** [comment]
Setting Department hours and ignoring operator status may help, but it would still be nice to have the operator status of each transfer department checked too.

Here is the embed code we use:
```
		  <script type="text/javascript">
		  var LHCChatOptions = {};
		  LHCChatOptions.opt = {widget_height:340,widget_width:300,popup_height:520,popup_width:500,domain:'example.com'};
		  (function() {
		  var po = document.createElement('script'); po.type = 'text/javascript'; po.async = true;
		  var refferer = (document.referrer) ? encodeURIComponent(document.referrer.substr(document.referrer.indexOf('://')+1)) : '';
		  var location  = (document.location) ? encodeURIComponent(window.location.href.substring(window.location.protocol.length)) : '';
		  po.src = 'https://chat.example.com/index.php/chat/getstatus/(click)/internal/(position)/bottom_right/(ma)/br/(top)/350/(units)/pixels/(leaveamessage)/true?r='+refferer+'&l='+location;
		  var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(po, s);
		  })();
		  </script>
```

**Comment by remdex** [comment]
Yes, I understand the issue itself, just you will have to wait :)

**Comment by ipso** [comment]
Another way (though likely more work) to get the same result would be the ability to assign weights to each operator on a per department basis, so we could use a single "Support" department, and just make the Tier 1 operators lower weight than the Tier 2 operators, so the operator auto-assign code will prefer to pick the lower weighed operators first. Only if they are busy/unavailable would it move to higher weighed operators and try them next.

This would also be nice for training new staff members, so we could have new staff members with a higher weight and receive fewer chat requests than the experienced ones. 

This would actually be the preferred approach, as its much more flexible with less work to setup.

**Comment by remdex** [comment]
I see, but it sounds more like separate issue :)
